### DISCUSSION TITLE
Get the body of the gmail

### DISCUSSION THREAD

**Discussion by headwinds** [julien040/anyquery#15] [Q&A] [answered]
I thought I would try anyquery to help me get to inbox zero. I can query and those stubborn 100 emails I have in my inbox and returns everything except the body of each email.

This the column data that is included - just missing the email body!

```
|  uid   |  subject  |  sent_at  | received_at |   _from   |    to     | reply_to  |    cc     | bcc | message_id |   flags   | size  | folder 
```

Is it possible to also have the content included as a column?

I'm on a mac using zsh shell and this query works for me to get all my emails I send to myself to read later but rarely do

```
anyquery -q 'SELECT * FROM gmail_imap_emails WHERE `to` = `_from` LIMIT 100' --json > emails_sent_to_self.json
```

Maybe...I might need to make another sql query with the uid to get the body from a different table?!  So close... I get a nice json result set... just missing the body

**Comment by julien040** [comment]
Hi @headwinds,  
Sorry, but it's not currently possible to fetch the body of emails. However, I can add support for fetching bodies.

I'll look into it this weekend. In the meantime, check the [Nylas](https://anyquery.dev/integrations/nylas/) plugin; it allows you to fetch the body of emails. However, it's very slow (5 seconds per 20 emails).

**Reply by headwinds** [reply]
Much thanks @julien040 - for now I plan to use the Gmail API and will also look into your Nylas suggestion 

I can still use `anyquery` for the subject query (who is emailing me and about what) and then hopefully use the Gmail API for a deeper review on the contents 

If it is possible to eventually also see the body of the email through `anyquery` that sounds like an awesome feature to to me - it's also good to know if its simply not possible so I appreciate your efforts to find out either way - keep me posted!

**Comment by julien040** [comment, marked as answer]
Hey @headwinds, I've updated the `imap` plugin. You can now query emails body using the `imap_emails_body` table. 
To install the update, run:
```bash
anyquery registry refresh
anyquery plugin update imap
```

Example of a query:
```sql
-- List all emails containing a specific word in the body
SELECT * FROM imap_emails_body WHERE body LIKE '%meeting with John%';
```
